casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -97,15 +100,24 @@ object Genesis {
       walletsFile <- toFile[F](maybeWalletsPath, genesisPath.resolve("wallets.txt"))
       wallets     <- getWallets[F](walletsFile, maybeWalletsPath)
       bonds       <- getBonds[F](bondsFile, numValidators, genesisPath)
-      validators  = bonds.toSeq.map(Validator.tupled)
       genesisBlock <- createGenesisBlock(
                        runtimeManager,
                        Genesis(
                          shardId = shardId,
                          timestamp = timestamp,
                          wallets = wallets,
-                         proofOfStake = ProofOfStake(minimumBond, maximumBond, validators),
-                         faucet = faucet
+                         proofOfStake = ProofOfStake(
+    minimumBond = minimumBond,
+    maximumBond = maximumBond,
+    validators = bonds.toSeq.map(Validator.tupled)
+  ),
+                         faucet = faucet,
+                         genesisPk = Ed25519.newKeyPair._2,
+                         vaults = Traverse[List]
+    .traverse(bonds.keys.toList)(RevAddress.fromPublicKey)
+    .get
+    .map(Vault(_, 1000L)),
```

Comment:
This is obciously wrong, it must not be hardcoded. I thought the bonds file has the bond amount?

Reply:
Do we want validators to have exactly the amount they want to bond in their vaults? That means, once they're bonded, they can't run and pay for contracts.

Reply:
If the amount they're instantiated with is greater than their bonding amount, how much greater should it be? It seems like, either way, we have to choose an arbitrary quantity.

Reply:
This is production code for setting up genesis. Everything should add up to the last dime :D The rest of their funds is going to be in their vaults that weren't used for bonding.

Or, maybe it should be the case that each bonds entry should be followed by there being sufficient funds in the corresponding vault (specifiec in wallets file)

One thing is for sure, this cannot stay hardcoded / chosen arbitrarily.